docs/user-interface/handlers/customize.md

```diff
@@ -1,122 +1,316 @@
+- `HandlerChanging` is fired before an existing handler is removed from a cross-platform control, and before a new handler for the cross-platform control is created. This event indicates that the existing native control is about be removed from the cross-platform control, and therefore that any native events should be unwired and other cleanup performed. The `HandlerChangingEventArgs` object that accompanies this event has `OldHandler` and `NewHandler` properties, of type `IElementHandler`.
 
-The following example customizes the background color of every control in the app, on Android, to cyan:
+In addition to these events, each control also has an overridable `OnHandlerChanged` method that's invoked when the `HandlerChanged` event fires, and a `OnHandlerChanging` method that's invoked when the `HandlerChanging` event fires.
+
+## Customize a control with a mapper
+
+The .NET MAUI `Entry` is a single-line text input control, that implements the `IEntry` interface. On iOS, the `EntryHandler` maps the `Entry` to an iOS `UITextField`. On Android, the `Entry` is mapped to an `AppCompatEditText`, and on Windows the `Entry` is mapped to a `TextBox`:
+
+:::image type="content" source="customize-images/entry-handler.png" alt-text="Entry handler architecture." border="false":::
+
+The `EntryMapper`, in the `EntryHandler` class, maps the cross-platform control API to the native control API. This mapper can be modified to customize the control on each platform:
 
 ```csharp
 using Microsoft.Maui;
 using Microsoft.Maui.Controls;
+using Microsoft.Maui.Graphics;
 
-public partial class App : Application
+namespace HandlersDemos.Views
 {
-    public App()
+    public partial class CustomizeEntryPage : ContentPage
     {
-        InitializeComponent();
+        public CustomizeEntryPage()
+        {
+            InitializeComponent();
 
+            Microsoft.Maui.Handlers.EntryHandler.EntryMapper["MyCustomization"] = (handler, view) =>
+            {
 #if __ANDROID__
-        Microsoft.Maui.Handlers.ViewHandler.ViewMapper[nameof(IView.Background)] = (h, v) =>
-        {
-            (h.NativeView as Android.Views.View).SetBackgroundColor(Microsoft.Maui.Graphics.Colors.Cyan.ToNative());
-        };
+                handler.NativeView.SetBackgroundColor(Colors.Transparent.ToNative());
+#elif __iOS__
+                handler.NativeView.BorderStyle = UIKit.UITextBorderStyle.None;
+#elif WINDOWS
+                handler.NativeView.FontWeight = Microsoft.UI.Text.FontWeights.Thin;
 #endif
+            };
+        }
     }
 }
 ```
 
-### Remove Android Entry underline
+In this example, the `Entry` customization occurs in a page class. Therefore, all `Entry` controls on Android, iOS, and Windows will be customized once the `CustomizeEntryPage` is displayed. The following customization is performed by using compiler preprocessing directives:
```

Comment:
Am I correct in this assumption? On the other UI stacks I've used, you have to call Show or add the page to something else. Technically you could do `var page = new CustomizeEntryPage()` and that doesn't show the page, but the constructor has run and now the handler code has run.

```suggestion
In this example, the `Entry` customization occurs in a page class. Therefore, all `Entry` controls on Android, iOS, and Windows will be customized once an instance of the `CustomizeEntryPage` class is created, but before it's displayed. The following customization is performed by using compiler preprocessing directives:
```